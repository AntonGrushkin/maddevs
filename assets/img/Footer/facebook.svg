<svg width="7" height="14" viewBox="0 0 7 14" fill="none" xmlns="http://www.w3.org/2000/svg">
<path fill-rule="evenodd" clip-rule="evenodd" d="M0 4.7155H1.395V3.35875C1.395 2.761 1.41 1.83775 1.845 1.26625C2.3025 0.6625 2.93175 0.25 4.01175 0.25C5.77275 0.25 6.5145 0.50125 6.5145 0.50125L6.165 2.56975C6.165 2.56975 5.58375 2.401 5.04 2.401C4.49775 2.401 4.01175 2.596 4.01175 3.1375V4.7155H6.23775L6.0825 6.73525H4.01175V13.753H1.395V6.73525H0V4.7155Z" fill="#96969C"/>
</svg>
