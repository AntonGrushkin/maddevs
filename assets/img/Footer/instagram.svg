<svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">
<path fill-rule="evenodd" clip-rule="evenodd" d="M8 0.5C5.963 0.5 5.708 0.509 4.90775 0.545C4.10975 0.581 3.5645 0.7085 3.0875 0.89375C2.58659 1.08169 2.13295 1.37712 1.7585 1.75925C1.37737 2.13378 1.08229 2.58676 0.89375 3.08675C0.7085 3.5645 0.58175 4.109 0.545 4.90775C0.509 5.70725 0.5 5.963 0.5 8C0.5 10.037 0.509 10.2927 0.545 11.0922C0.58175 11.891 0.7085 12.4348 0.89375 12.9132C1.08229 13.4132 1.37737 13.8662 1.7585 14.2407C2.13295 14.6229 2.58659 14.9183 3.0875 15.1063C3.5645 15.2923 4.10975 15.419 4.90775 15.4542C5.708 15.491 5.963 15.4993 8 15.4993C10.037 15.4993 10.292 15.491 11.0922 15.4542C11.8902 15.419 12.4355 15.2923 12.9125 15.1063C13.4134 14.9183 13.8671 14.6229 14.2415 14.2407C14.6226 13.8662 14.9177 13.4132 15.1063 12.9132C15.2915 12.4355 15.4183 11.891 15.455 11.0922C15.491 10.2927 15.5 10.037 15.5 8C15.5 5.963 15.491 5.70725 15.455 4.90775C15.4183 4.109 15.2915 3.5645 15.1063 3.08675C14.9177 2.58676 14.6226 2.13378 14.2415 1.75925C13.8671 1.37712 13.4134 1.08169 12.9125 0.89375C12.4355 0.7085 11.8902 0.581 11.0922 0.54575C10.292 0.5075 10.037 0.5 8 0.5ZM8 1.8515C10.0025 1.8515 10.2395 1.859 11.0308 1.895C11.762 1.928 12.1588 2.05025 12.4235 2.153C12.773 2.2895 13.0235 2.45225 13.2853 2.714C13.5478 2.9765 13.7105 3.227 13.8463 3.5765C13.9497 3.84125 14.0712 4.238 14.105 4.96925C14.141 5.7605 14.1485 5.99675 14.1485 7.99925C14.1485 10.0017 14.141 10.2395 14.105 11.0308C14.0712 11.762 13.9497 12.1587 13.8463 12.4227C13.7105 12.773 13.5478 13.0228 13.2853 13.2853C13.0433 13.5342 12.7489 13.7261 12.4235 13.847C12.1588 13.9497 11.762 14.072 11.0308 14.105C10.2395 14.141 10.0025 14.1485 8 14.1485C5.9975 14.1485 5.7605 14.141 4.96925 14.105C4.238 14.072 3.84125 13.9497 3.5765 13.847C3.25113 13.7264 2.95674 13.5347 2.71475 13.286C2.46567 13.0437 2.27398 12.7488 2.15375 12.4227C2.05025 12.1587 1.92875 11.762 1.895 11.0308C1.859 10.2395 1.8515 10.0025 1.8515 8C1.8515 5.9975 1.859 5.7605 1.895 4.96925C1.92875 4.238 2.05025 3.84125 2.15375 3.57725C2.2895 3.227 2.45225 2.97725 2.71475 2.714C2.95674 2.46528 3.25113 2.27364 3.5765 2.153C3.84125 2.05025 4.238 1.928 4.96925 1.895C5.7605 1.859 5.9975 1.8515 8 1.8515Z" fill="#96969C"/>
<path fill-rule="evenodd" clip-rule="evenodd" d="M7.99969 10.4999C7.66782 10.5057 7.33813 10.4453 7.02986 10.3223C6.72159 10.1992 6.4409 10.016 6.20417 9.78339C5.96745 9.55073 5.77943 9.27326 5.65109 8.96716C5.52274 8.66106 5.45664 8.33247 5.45664 8.00056C5.45664 7.66864 5.52274 7.34005 5.65109 7.03395C5.77943 6.72785 5.96745 6.45038 6.20417 6.21772C6.4409 5.98506 6.72159 5.80188 7.02986 5.67886C7.33813 5.55583 7.66782 5.49543 7.99969 5.50118C8.65507 5.51253 9.27976 5.78085 9.73922 6.24834C10.1987 6.71583 10.4561 7.34508 10.4561 8.00056C10.4561 8.65603 10.1987 9.28528 9.73922 9.75277C9.27976 10.2203 8.65507 10.4886 7.99969 10.4999ZM7.99969 4.14893C6.97827 4.14893 5.99869 4.55469 5.27644 5.27694C4.55419 5.99919 4.14844 6.97877 4.14844 8.00018C4.14844 9.0216 4.55419 10.0012 5.27644 10.7234C5.99869 11.4457 6.97827 11.8514 7.99969 11.8514C9.0211 11.8514 10.0007 11.4457 10.7229 10.7234C11.4452 10.0012 11.8509 9.0216 11.8509 8.00018C11.8509 6.97877 11.4452 5.99919 10.7229 5.27694C10.0007 4.55469 9.0211 4.14893 7.99969 4.14893ZM12.9032 3.99668C12.9032 4.11487 12.8799 4.2319 12.8347 4.3411C12.7895 4.45029 12.7232 4.5495 12.6396 4.63308C12.556 4.71665 12.4568 4.78294 12.3476 4.82817C12.2384 4.8734 12.1214 4.89668 12.0032 4.89668C11.885 4.89668 11.768 4.8734 11.6588 4.82817C11.5496 4.78294 11.4504 4.71665 11.3668 4.63308C11.2832 4.5495 11.2169 4.45029 11.1717 4.3411C11.1265 4.2319 11.1032 4.11487 11.1032 3.99668C11.1032 3.75798 11.198 3.52907 11.3668 3.36028C11.5356 3.1915 11.7645 3.09668 12.0032 3.09668C12.2419 3.09668 12.4708 3.1915 12.6396 3.36028C12.8084 3.52907 12.9032 3.75798 12.9032 3.99668Z" fill="#96969C"/>
</svg>
