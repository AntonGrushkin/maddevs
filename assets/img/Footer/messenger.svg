<svg width="16" height="17" viewBox="0 0 16 17" fill="none" xmlns="http://www.w3.org/2000/svg">
<g clip-path="url(#clip0)">
<path fill-rule="evenodd" clip-rule="evenodd" d="M2.94679 16.4802V13.6342C1.14843 12.2777 0 10.2119 0 7.89823C0 3.81228 3.58171 0.5 8 0.5C12.4186 0.5 16 3.81228 16 7.89823C16 11.9839 12.4186 15.2965 8 15.2965C7.17571 15.2965 6.38014 15.1811 5.63171 14.9673L2.94679 16.4802ZM2.83203 10.3996L7.14639 5.87207L9.2171 7.96991L13.1 5.87207L8.80996 10.3996L6.75846 8.26442L2.83203 10.3996Z" fill="#96969C"/>
</g>
<defs>
<clipPath id="clip0">
<rect width="16" height="16" fill="white" transform="translate(0 0.5)"/>
</clipPath>
</defs>
</svg>
