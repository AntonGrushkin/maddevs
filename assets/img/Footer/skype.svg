<svg width="16" height="17" viewBox="0 0 16 17" fill="none" xmlns="http://www.w3.org/2000/svg">
<g clip-path="url(#clip0)">
<path d="M4.71875 9.75C7.22129 9.75 9.25 7.72129 9.25 5.21875C9.25 2.71621 7.22129 0.6875 4.71875 0.6875C2.21621 0.6875 0.1875 2.71621 0.1875 5.21875C0.1875 7.72129 2.21621 9.75 4.71875 9.75Z" fill="#96969C"/>
<path d="M11.2812 16.3125C13.7838 16.3125 15.8125 14.2838 15.8125 11.7812C15.8125 9.27871 13.7838 7.25 11.2812 7.25C8.77871 7.25 6.75 9.27871 6.75 11.7812C6.75 14.2838 8.77871 16.3125 11.2812 16.3125Z" fill="#96969C"/>
<path d="M8 16C12.1421 16 15.5 12.6421 15.5 8.5C15.5 4.35786 12.1421 1 8 1C3.85786 1 0.5 4.35786 0.5 8.5C0.5 12.6421 3.85786 16 8 16Z" fill="#96969C"/>
<path fill-rule="evenodd" clip-rule="evenodd" d="M6.05423 8.35769C5.76103 8.16454 5.51641 7.90626 5.33947 7.60301C5.16851 7.28279 5.08411 6.9235 5.09462 6.56066C5.08161 6.10017 5.23873 5.65102 5.53596 5.29905C5.84478 4.9456 6.24173 4.68028 6.68642 4.53009C7.17016 4.35927 7.67984 4.27352 8.19285 4.27663C8.52857 4.27345 8.86394 4.29917 9.19524 4.35352C9.43223 4.39016 9.66499 4.45026 9.89009 4.53295C10.0969 4.60298 10.2835 4.72221 10.434 4.88037C10.545 5.01047 10.6047 5.17651 10.602 5.34745C10.606 5.43625 10.5925 5.52496 10.5622 5.60852C10.5319 5.69208 10.4854 5.76885 10.4255 5.83445C10.3687 5.89491 10.2997 5.94266 10.2232 5.97456C10.1466 6.00647 10.0641 6.02181 9.98123 6.01958C9.8671 6.01874 9.75447 5.99348 9.65091 5.9455C9.39595 5.83138 9.1333 5.73528 8.8649 5.65789C8.61776 5.59278 8.363 5.56117 8.10745 5.56392C7.77463 5.55157 7.44532 5.63562 7.15913 5.80597C7.03825 5.87938 6.93932 5.98393 6.8727 6.10869C6.80607 6.23344 6.77419 6.3738 6.78041 6.5151C6.77862 6.70044 6.84909 6.8792 6.97687 7.01346C7.12882 7.17102 7.30836 7.29939 7.50658 7.39224C7.72872 7.50242 8.05999 7.64861 8.5004 7.83083C8.54922 7.84625 8.59683 7.8653 8.64281 7.8878C9.07258 8.05776 9.48462 8.26952 9.873 8.52003C10.1798 8.72008 10.4385 8.98559 10.6305 9.2975C10.8195 9.62516 10.9141 9.99872 10.9039 10.3768C10.9198 10.84 10.787 11.296 10.5251 11.6783C10.2584 12.0402 9.88947 12.314 9.46581 12.4644C8.95531 12.6467 8.41588 12.7345 7.87392 12.7235C7.12913 12.7497 6.38787 12.6097 5.70402 12.3134C5.52902 12.2356 5.37558 12.1164 5.25693 11.966C5.15858 11.8283 5.1076 11.6624 5.11168 11.4932C5.10677 11.404 5.12129 11.3147 5.15422 11.2316C5.18715 11.1485 5.23772 11.0736 5.30244 11.0119C5.43828 10.893 5.61466 10.8308 5.79508 10.8382C5.99589 10.841 6.19312 10.8918 6.3703 10.9863C6.59046 11.0965 6.76418 11.1791 6.89146 11.2341C7.03686 11.295 7.18745 11.3427 7.34142 11.3765C7.53121 11.4182 7.72512 11.4382 7.91943 11.4363C8.26298 11.4607 8.60484 11.3695 8.89054 11.1771C8.99752 11.0934 9.08302 10.9855 9.13996 10.8621C9.19691 10.7388 9.22366 10.6037 9.218 10.468C9.21842 10.2754 9.14507 10.0899 9.013 9.94966C8.83848 9.76987 8.63426 9.62151 8.40932 9.51112C8.14349 9.37064 7.76947 9.19977 7.28727 8.99849C6.85588 8.82661 6.4428 8.61194 6.05422 8.35769" fill="black"/>
<path fill-rule="evenodd" clip-rule="evenodd" d="M6.13477 0.916016C5.93663 0.849677 5.73404 0.797463 5.52852 0.759766C4.29462 1.15575 3.17297 1.84032 2.25664 2.75664C1.34032 3.67297 0.655754 4.79462 0.259766 6.02852C0.297469 6.23403 0.349683 6.43663 0.416016 6.63477C0.471415 6.81416 0.541441 6.9887 0.62539 7.15664C0.896838 5.65673 1.61983 4.27548 2.69766 3.19766C3.77548 2.11983 5.15673 1.39684 6.65664 1.12539C6.4887 1.04145 6.31415 0.971425 6.13477 0.916016Z" fill="black"/>
<path fill-rule="evenodd" clip-rule="evenodd" d="M6.13477 0.916016C4.75079 1.25736 3.48646 1.97057 2.47852 2.97852C1.47057 3.98646 0.757362 5.25079 0.416016 6.63477C0.471415 6.81416 0.541441 6.9887 0.62539 7.15664C0.896838 5.65673 1.61983 4.27548 2.69766 3.19766C3.77548 2.11983 5.15673 1.39684 6.65664 1.12539C6.4887 1.04145 6.31415 0.971425 6.13477 0.916016Z" fill="black"/>
<path fill-rule="evenodd" clip-rule="evenodd" d="M15.5844 10.3656C15.529 10.1862 15.4589 10.0117 15.375 9.84375C15.1036 11.3437 14.3806 12.7249 13.3027 13.8027C12.2249 14.8806 10.8437 15.6036 9.34375 15.875C9.51169 15.9589 9.68623 16.029 9.86563 16.0844C10.0638 16.1507 10.2664 16.2029 10.4719 16.2406C11.7058 15.8446 12.8274 15.1601 13.7437 14.2437C14.6601 13.3274 15.3446 12.2058 15.7406 10.9719C15.7029 10.7664 15.6507 10.5638 15.5844 10.3656Z" fill="black"/>
<path fill-rule="evenodd" clip-rule="evenodd" d="M15.375 9.84375C15.1036 11.3437 14.3806 12.7249 13.3027 13.8027C12.2249 14.8806 10.8437 15.6036 9.34375 15.875C9.51169 15.9589 9.68623 16.029 9.86563 16.0844C11.2496 15.743 12.5139 15.0298 13.5219 14.0219C14.5298 13.0139 15.243 11.7496 15.5844 10.3656C15.529 10.1862 15.4589 10.0117 15.375 9.84375Z" fill="black"/>
</g>
<defs>
<clipPath id="clip0">
<rect y="0.5" width="16" height="16" fill="white"/>
</clipPath>
</defs>
</svg>
