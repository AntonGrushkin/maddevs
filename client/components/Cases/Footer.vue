<template>
  <section class="case_footer">
    <div class="case_logotype nambafood"></div>
    <h2 class="case_title case_title_h2">
      <slot></slot>
    </h2>
    <button type="button" class="case_next-project-button">Read next project</button>
  </section>
</template>

<script>
export default {
  name: 'Header'
};
</script>

<style scoped lang="scss">
@import "../../assets/styles/cases/mixins";

.case {
  &_footer {
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
    padding-top: 85px;
    padding-bottom: 115px;
    background-color: $bgcolor--white-transparent;
  }

  &_logotype {
    @include nambafood-footer;
    width: 115px;
    height: 103px;
    margin-bottom: 41px;
  }

  &_title {
    margin-bottom: 51px;
    text-align: center;
  }

  &_next-project-button {
    padding: 11px 28px;
    background-color: $bgcolor--black;
    border: 0;
    cursor: pointer;
    @include default_text($text-color--white-primary, 17px, 166%, -0.035em, normal);
  }
}
</style>
