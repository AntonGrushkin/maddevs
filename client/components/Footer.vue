<template>
  <footer :class="`footer ${currentPageName}`">
    <div class="container">
			<div class="footer__content row">
				<div class="footer__text-content col-xl-7 col-lg-8 col-md-7">
					<div class="footer__title-wrap">
						<h2 class="footer__title title">Get In Touch</h2>
						<div class="footer__mail-wrapper">
							<a class="footer__mailto-link" href="mailto:team@maddevs.io">team@maddevs.io</a>
							<span class="footer__underline" />
						</div>
					</div>
          <footerContacts />
					<footerSocialNetworkBar class="footer__social-network"/>
					<div class="footer__links-wrapper">
						<p class="footer__company-name">Â© Mad Devs - {{ currentYear }}</p>
						<div class="footer__links">
							<a href="/gdpr" target="_blank" class="footer__link">GDPR Compliance Commitment</a>
							<a href="/privacy" target="_blank" class="footer__link">Privacy Policy</a>
							<a href="/nda" target="_blank" class="footer__link">Non-Disclosure Agreement (NDA)</a>
						</div>
					</div>
				</div>
				<div class="footer__form-wrap col-xl-5 col-lg-4 col-md-5">
					<footerForm />
				</div>
				<footerBottomContentMobile />
			</div>
    </div>
  </footer>
</template>

<script>
import footerForm from '@/components/Footer/footer-form';
import footerContacts from '@/components/Footer/footer-contacts';
import footerSocialNetworkBar from '@/components/Footer/footer-social-network';
import footerBottomContentMobile from '@/components/Footer/footer-bottom-content-mobile';

export default {
  name: 'Footer',
  components: {
    footerForm,
    footerContacts,
    footerSocialNetworkBar,
    footerBottomContentMobile
  },
  mounted() {
    if ($nuxt.$route.name) {
      this.currentPageName = $nuxt.$route.name;
    }
  },
  data() {
    return {
      currentYear: new Date().getFullYear(),
      currentPageName: ''
    };
  }
};
</script>

<style lang="scss" scoped>
  @import '../assets/styles/vars';

	.footer {
		padding-top: 65px;
		padding-bottom: 59px;

		&__title {
		  color: $section-title-white;
		}

		&__text-content {
			padding: 0
		}

		&__mail-wrapper {
			width: max-content;
		}

		&__mailto-link {
			color: $text-color--red;
			font-size: 48px;
			font-family: 'Poppins-Bold';
			text-decoration: none;
			letter-spacing: -1px;
		}

		&__underline {
			width: 100%;
			height: 1px;
			display: block;
			margin-top: -15px;
			background-color: $bgcolor--red;
		}

		&__links-wrapper {
			width: max-content;
			margin-top: 79px;
		}

		&__links {
			display: flex;
		}

		&__link,
		&__company-name {
			font-size: 15px;
			font-family: 'Formular-Regular';
			color: $text-color--grey;
			letter-spacing:  -0.02em;
		}

		&__company-name {
			margin-bottom: 6px;
		}

		&__link {
			padding-left: 26px;
			text-decoration: underline;
			text-decoration-color: $footer--text-decoration-color;

			&:first-child {
				padding-left: 0;
			}
		}

		&__form-wrap {
			margin-top: 18px;
		}
	}

	.gdpr,
	.privacy,
	.nda {
		padding-top: 109px;
		margin-top: 90px;
		border-top: 1px solid $border-color--grey-dark;
	}

	@media only screen and (max-width: 1024px) {
		.footer {
			&__mailto-link {
				font-size: 32px;
			}

			&__underline {
				margin-top: -11px;
			}

			&__links-wrapper {
				margin-top: 45px;
			}

			&__link,
			&__company-name {
				font-size: 12px;
			}
		}
	}

	@media only screen and (max-width: 767px) {
		.footer {
			&__links-wrapper,
			&__social-network {
				display: none;
			}

			&__form-wrap {
				padding-top: 39px;
			}
		}
	}

</style>
