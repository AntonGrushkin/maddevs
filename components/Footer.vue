<template>
  <footer class="footer">
    <div class="container">
			<div class="footer__content">
				<div class="footer__text-content">
					<div class="footer__title-wrap">
						<h2 class="footer__main-title main-title">Get In Touch</h2>
						<a class="footer__mailto-link" href="mailto:team@maddevs.io">team@maddevs.io</a>
					</div>
          <footerContacts />
				</div>
				<div class="footer__form-wrap">
					<footerForm />
				</div>
			</div>
			<div class="footer__bottom-links-line">
				<div class="footer__left-bottom_links">
					<p class="footer__company-name">Â© Mad Devs - {{ currentYear }}</p>
					<a href="https://maddevs.io/gdpr/" class="footer__bottom-link">GDPR Compliance Commitment</a>
					<a href="#" class="footer__bottom-link">Privacy Policy</a>
					<a href="/nda" class="footer__bottom-link">Non-Disclosure Agreement (NDA)</a>
				</div>
				<footerSocialNetworkBar />
			</div>
    </div>
  </footer>
</template>

<script>
import footerForm from '@/components/Footer/footer-form';
import footerContacts from '@/components/Footer/footer-contacts';
import footerSocialNetworkBar from '@/components/Footer/footer-social-network';

export default {
  name: 'Footer',
  components: {
    footerForm,
    footerContacts,
    footerSocialNetworkBar
  },
  data() {
    return {
      currentYear: new Date().getFullYear()
    };
  }
};
</script>

<style lang="scss" scoped>
  @import '../assets/styles/vars';

	.footer {
		padding-top: 129px;
		padding-bottom: 87px;

		&__content {
			display: flex;
    	justify-content: space-between;
			padding-bottom: 57px;
		}

		&__main-title {
			text-shadow: $text-shadow--white;
		}

		&__mailto-link {
			color: $text-color--red;
			font-size: 70px;
			font-family: 'Hoves-Bold';
			text-decoration: none;
			border-bottom: 1px solid;
		}

		&__bottom-links-line {
			display: flex;
			justify-content: space-between;
			padding-top: 23px;
			border-top: 1px solid $footer--border-color--grey-light;
		}

		&__left-bottom_links {
			display: flex;
		}

		&__bottom-link,
		&__company-name {
			font-size: 16px;
			font-family: 'Hoves-Regular';
			color: $text-color--grey;
		}

		&__bottom-link {
			padding-left: 26px;
			text-decoration: none;

			&:first-child {
				padding-left: 30px;
			}

			&:nth-child(3) {
				text-decoration: underline;
				text-decoration-color: $footer--text-decoration-color;
			}
		}
	}

	@media only screen and (max-width: 1320px) {
		.footer {
			&__bottom-link,
			&__company-name {
				font-size: 13px;
			}
		}
	}

	@media only screen and (max-width: 1180px) {
		.footer {
			&__mailto-link {
				font-size: 55px;
			}
		}
	}

	@media only screen and (max-width: 1150px) {
		.footer {
			&__bottom-link {
				padding-left: 15px;
			}
		}
	}

	@media only screen and (max-width: 1024px) {
		.footer {
			&__content {
				flex-direction: column;
			}

			&__bottom-links-line {
				flex-direction: column-reverse;
				padding-left: 0;
			}

			&__left-bottom_links {
				padding-top: 20px;
			}
		}
	}

	@media only screen and (max-width: 920px) {
		.footer {
			&__main-title {
				font-size: 120px;
			}
		}
	}

	@media only screen and (max-width: 870px) {
		.footer {
			&__main-title {
				font-size: 100px;
			}
		}
	}

	@media only screen and (max-width: 768px) {
		.footer {
			&__mailto-link {
				font-size: 36px;
			}

			&__left-bottom_links {
				display: grid;
				grid-template-columns: 1fr;
			}

			&__bottom-link {
				padding-top: 8px;
				padding-left: 0;

				&:nth-child(3) {
					grid-row-start: 3;
				}
			}

			&__company-name {
				grid-row-start: 4;
				padding-top: 35px;
			}
		}
	}

	@media only screen and (max-width: 728px) {
		.footer {
			&__main-title {
				font-size: 75px;
			}
		}
	}

	@media only screen and (max-width: 600px) {
		.footer {
			&__main-title {
				font-size: 75px;
			}
		}
	}

	@media only screen and (max-width: 520px) {
		.footer {
			&__main-title {
				font-size: 65px;
			}
		}
	}

	@media only screen and (max-width: 440px) {
		.footer {
			&__main-title {
				font-size: 55px;
			}
		}
	}

	@media only screen and (max-width: 370px) {
		.footer {
			&__main-title {
				font-size: 45px;
			}

			&__mailto-link {
				font-size: 25px;
			}
		}
	}
</style>
