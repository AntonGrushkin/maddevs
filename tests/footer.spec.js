import { mount } from '@vue/test-utils';
import Footer from '@/components/Footer';

describe('Footer', () => {
  let wrapper;

  beforeEach(() => {
    global.$nuxt = {
      $route: {
        name: '/'
      }
    };
    wrapper = mount(Footer, {
      stubs: ['ValidationProvider', 'ValidationObserver', 'router-link']
    });
  });

  // ------ IMPORTANT ----- //
  test('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
  // --------------------- //

  it('correctly sets the route when mounted', () => {
    expect(wrapper.vm.$data.currentPageName).toBe('/');
  });

  test('correctly sets the current pages name and class', () => {
    const $route = {
      path: '/gdpr'
    };

    wrapper = mount(Footer, {
      mocks: {
        $route
      }
    });

    wrapper.vm.$data.currentPageName = 'gdpr';
    expect(wrapper.vm.$route.path).toBe('/gdpr');
    expect(wrapper.vm.$data.currentPageName).toBe('gdpr');

    let wrapperClass = wrapper.findAll('.gdpr');
    expect(wrapperClass).toEqual({
      selector: '.gdpr'
    });
  });
});
