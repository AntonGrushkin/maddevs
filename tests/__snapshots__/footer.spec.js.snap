// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<footer
  class="footer"
>
  <div
    class="container"
  >
    <div
      class="footer__content"
    >
      <div
        class="footer__text-content"
      >
        <div
          class="footer__title-wrap"
        >
          <h2
            class="footer__main-title main-title"
          >
            Get In Touch
          </h2>
           
          <a
            class="footer__mailto-link"
            href="mailto:team@maddevs.io"
          >
            team@maddevs.io
          </a>
        </div>
         
        <div
          class="footer-contacts"
        >
          <div
            class="footer-contacts__call-us"
          >
            <h4
              class="footer-contacts__title"
            >
              Call us:
            </h4>
             
            <ul
              class="footer-contacts__phones-list"
            >
              <li>
                <a
                  class="footer-contacts__phone-item"
                  href="tel:+1 888 401 3401"
                >
                  <img
                    alt="United States"
                    src="../../assets/img/Flags/united-states.svg"
                  />
                  
					+1 888 401 3401
				
                </a>
              </li>
               
              <li>
                <a
                  class="footer-contacts__phone-item"
                  href="tel:+1 208 473 2904"
                >
                  <img
                    alt="United States"
                    src="../../assets/img/Flags/united-states.svg"
                  />
                  
					+1 208 473 2904
				
                </a>
              </li>
               
              <li>
                <a
                  class="footer-contacts__phone-item"
                  href="tel:+1 888 401 3401"
                >
                  <img
                    alt="United Kingdom"
                    src="../../assets/img/Flags/united-kingdom.svg"
                  />
                  
					+1 888 401 3401
				
                </a>
              </li>
               
              <li>
                <a
                  class="footer-contacts__phone-item"
                  href="tel:+1 208 473 2904"
                >
                  <img
                    alt="United Kingdom"
                    src="../../assets/img/Flags/united-kingdom.svg"
                  />
                  
					+1 208 473 2904
				
                </a>
              </li>
            </ul>
          </div>
           
          <div
            class="footer-contacts__message-us"
          >
            <h4
              class="footer-contacts__title"
            >
              Message us:
            </h4>
             
            <ul
              class="footer-contacts__messengers-list"
            >
              <li>
                <a
                  class="footer-contacts__messenger-item"
                  href="https://t.me/mun_tamara/"
                  target="_blank"
                >
                  <img
                    alt="Telegram"
                    src="../../assets/img/Footer/telegram.svg"
                  />
                  
					Telegram
				
                </a>
              </li>
               
              <li>
                <a
                  class="footer-contacts__messenger-item"
                  href="http://wa.me/996555771481"
                  target="_blank"
                >
                  <img
                    alt="Whatsapp"
                    src="../../assets/img/Footer/whatsapp.svg"
                  />
                  
					Whatsapp
				
                </a>
              </li>
               
              <li>
                <a
                  class="footer-contacts__messenger-item"
                  href="#"
                >
                  <img
                    alt="Signal"
                    src="../../assets/img/Footer/signal.svg"
                  />
                  
					Signal
				
                </a>
              </li>
               
              <li>
                <a
                  class="footer-contacts__messenger-item"
                  href="#"
                >
                  <img
                    alt="WeChat"
                    src="../../assets/img/Footer/wechat.svg"
                  />
                  
					WeChat
				
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
       
      <div
        class="footer__form-wrap"
      >
        <form
          class="footer-form form"
        >
          <div
            class="fields-list"
          >
            <div
              class="field-item"
            >
              <p
                class="field-name"
              >
                First name and surname
              </p>
               
              <input
                class="entry-field"
                placeholder="John Smith"
                type="text"
              />
            </div>
             
            <div
              class="field-item"
            >
              <p
                class="field-name required"
              >
                Work email
              </p>
               
              <input
                class="entry-field"
                placeholder="your@company.co|"
                required="required"
                type="email"
              />
            </div>
             
            <div
              class="field-item"
            >
              <p
                class="field-name"
              >
                Project Info
              </p>
               
              <input
                class="entry-field"
                placeholder="Describe your project..."
                type="text"
              />
            </div>
          </div>
           
          <div
            class="form-checkboxes"
          >
            <label
              class="form-checkbox-label required"
              for="privacy-policy"
            >
              I confirm that I have read and accepted Mad Devs’ 
              <br />
               Privacy Policy
			
              <input
                class="form-checkbox-input"
                id="privacy-policy"
                name="Privacy policy"
                type="checkbox"
              />
               
              <span
                class="checkmark"
              />
            </label>
             
            <label
              class="form-checkbox-label"
              for="marketing-communications"
            >
              I agree to get Mad Devs’ discount offers and other 
              <br />
               marketing communications
			
              <input
                class="form-checkbox-input"
                id="marketing-communications"
                name="Marketing communications"
                type="checkbox"
              />
               
              <span
                class="checkmark"
              />
            </label>
          </div>
           
          <button
            class="show-popup_button red"
          >
            
  Order a project now

          </button>
        </form>
      </div>
    </div>
     
    <div
      class="footer__bottom-links-line"
    >
      <div
        class="footer__left-bottom_links"
      >
        <p
          class="footer__company-name"
        >
          © Mad Devs - 2020
        </p>
         
        <a
          class="footer__bottom-link"
          href="https://maddevs.io/gdpr/"
        >
          GDPR Compliance Commitment
        </a>
         
        <a
          class="footer__bottom-link"
          href="#"
        >
          Privacy Policy
        </a>
         
        <a
          class="footer__bottom-link"
          href="https://maddevs.io/nda/"
        >
          Non-Disclosure Agreement (NDA)
        </a>
      </div>
       
      <div
        class="footer-social-network"
      >
        <a
          class="footer-social-network__social-newtork_link"
          href="https://github.com/maddevsio"
          target="_blank"
        >
          <img
            alt="Mad Devs GitHub"
            class="footer-social-network__github"
            src="../../assets/img/Footer/github.svg"
          />
        </a>
         
        <a
          class="footer-social-network__social-newtork_link"
          href="https://www.facebook.com/maddevsio"
          target="_blank"
        >
          <img
            alt="Mad Devs Facebook"
            class="footer-social-network__facebook"
            src="../../assets/img/Footer/facebook.svg"
          />
        </a>
         
        <a
          class="footer-social-network__social-newtork_link"
          href="https://www.instagram.com/maddevsio/"
          target="_blank"
        >
          <img
            alt="Mad Devs Instagram"
            class="footer-social-network__instagram"
            src="../../assets/img/Footer/instagram.svg"
          />
        </a>
         
        <a
          class="footer-social-network__social-newtork_link"
          href="https://blog.maddevs.io/"
          target="_blank"
        >
          <img
            alt="Mad Devs Medium"
            class="footer-social-network__medium"
            src="../../assets/img/Footer/medium.svg"
          />
        </a>
         
        <a
          class="footer-social-network__social-newtork_link"
          href="https://twitter.com/MadDevsIO"
          target="_blank"
        >
          <img
            alt="Mad Devs Twitter"
            class="footer-social-network__twitter"
            src="../../assets/img/Footer/twitter.svg"
          />
        </a>
         
        <a
          class="footer-social-network__social-newtork_link"
          href="https://www.slideshare.net/maddevs/presentations"
          target="_blank"
        >
          <img
            alt="Mad Devs Slideshare"
            class="footer-social-network__slideshare"
            src="../../assets/img/Footer/slideshare.svg"
          />
        </a>
         
        <a
          class="footer-social-network__social-newtork_link"
          href="https://www.youtube.com/channel/UCF6liiudpZq6AQe6hXVEW7w"
          target="_blank"
        >
          <img
            alt="Mad Devs YouTube"
            class="footer-social-network__youtube"
            src="../../assets/img/Footer/youtube.svg"
          />
        </a>
         
        <a
          class="footer-social-network__social-newtork_link"
          href="https://www.behance.net/maddevs"
          target="_blank"
        >
          <img
            alt="Mad Devs Behance"
            class="footer-social-network__behance"
            src="../../assets/img/Footer/behance.svg"
          />
        </a>
      </div>
    </div>
  </div>
</footer>
`;
